### PR TITLE
Update JSON-API & Recursive documentation

### LINKED ISSUE #3742: Update docs to show new JSON API endpoints
Steps:

1. Enumerate all endpoints that return JSON (basically update this list here: https://docs.ordinals.com/guides/explorer.html#json-api)
2. Add short description 
3. Add an example `curl` command with JSON response.

We should make a new section under Guides and call it API and pull out the JSON API stuff from the Explorer section.

### PR DESCRIPTION
**Pull request by cryptoni9n**
fixes #3742

draft mode for now while I look for typos and test the links and menus

I'm not confident I have all of the endpoints captured here yet.  Please comment if you see any missing.

### CODE REVIEW

**Comment by cryptoni9n** [comment]
I think this may be ready for a first review now.

**Comment by raphjaph** [comment]
This looks great so far! 

I made some changes so that the examples can be expanded underneath. It's based on [this](https://gist.github.com/azagniotov/a4b16faf0febd12efbc6c3d7370383a6). If you could transform the other things into that format, that would be great!

**Comment by raphjaph** [comment]
Yes, this looks great!

I've made some tweaks to the first two endpoints, if you could apply those to the rest as well. 

Also could you apply the same section to the Recursion section of the docs and update the examples as well? I think we're missing some. Thanks!

**Comment by cryptoni9n** [comment]
> Yes, this looks great!
> 
> I've made some tweaks to the first two endpoints, if you could apply those to the rest as well.

sure, I've applied those changes throughout.

> Also could you apply the same template to the Recursion section of the docs and update the examples as well? I think we're missing some. Thanks!

ok - should I make a new PR or would you like it in this one?

**Comment by cryptoni9n** [comment]
close/reopen to restart faulty tests

**Comment by raphjaph** [comment]
> ok - should I make a new PR or would you like it in this one?

You can put it in this one

File: docs/src/SUMMARY.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ Summary
   - [Settings](guides/settings.md)
   - [Teleburning](guides/teleburning.md)
   - [Testing](guides/testing.md)
+  - [API](guides/api.md)
```
Comment (raphjaph) [inline docs/src/SUMMARY.md:32]: Can we put this as the first entry to the guides section?

File: docs/src/inscriptions/recursion.md [hunk 2]
```diff
@@ -30,80 +30,124 @@ Recursion has a number of interesting use-cases:
...
+    <code>GET</code>
+    <code><b>/content/&lt;INSCRIPTION_ID&gt;</b></code>
+ </summary>
 
-- `/blockheight`: latest block height.
-- `/blockhash`: latest block hash.
-- `/blockhash/<HEIGHT>`: block hash at given block height.
-- `/blocktime`: UNIX time stamp of latest block.
+### Description
+the content of the inscription with `<INSCRIPTION_ID>`
```
Comment (raphjaph) [inline docs/src/inscriptions/recursion.md:43]:
```suggestion
The content of the inscription with `<INSCRIPTION_ID>`
```

File: docs/src/inscriptions/recursion.md [hunk 3]
```diff
@@ -30,80 +30,124 @@ Recursion has a number of interesting use-cases:
...
+<details>
+ <summary>
+    <code>GET</code>
+    <code><b>/r/blockhash/&lt;HEIGHT&gt;</b></code>
+ </summary>
 
-```json
-777000
+### Description
+block hash at given block height in text.
```
Comment (raphjaph) [inline docs/src/inscriptions/recursion.md:79]:
```suggestion
Block hash at given block height in text.
```

**Review by raphjaph — CHANGES_REQUESTED** [review]
Looking sweet! 

Could you capitalize the the first word in all of the descriptions in `recursion.md`.

While we're at it maybe also add the required flags for endpoint to be active just somewhere in the description. If it doesn't require any specific flags then it doesn't have to say anything. The basic config flags I can think about are `--index-addresses`, `--index-sats` and `--index-runes`.

File: docs/src/SUMMARY.md (hunk 1 above)
Comment (cryptoni9n) [reply docs/src/SUMMARY.md:32]: sure thing

File: docs/src/inscriptions/recursion.md (hunk 2 above)
Comment (cryptoni9n) [reply docs/src/inscriptions/recursion.md:43]: will do

File: docs/src/inscriptions/recursion.md (hunk 3 above)
Comment (cryptoni9n) [reply docs/src/inscriptions/recursion.md:79]: will do

**Review by raphjaph — APPROVED** [review]
This looks sweet!

Made some small tweaks so that it include the recursive endpoints on the API page as well. Let me know if it looks good and then I'll merge.

**Comment by cryptoni9n** [comment]
thanks Raph! I think it looks good, but when I do the rich diff display, the include doesn't look like it is rendering correctly.  Perhaps the viewer isn't capable?  Not sure...
image

Also, I saw one other thing that needed to be removed so I took it out.

**Comment by raphjaph** [comment]
The include statement is for the `mdbook` builder. You can run it and display the docs with `just serve-docs`

**Pull request merged**